### PR TITLE
Gainsightpx

### PR DESCRIPTION
**Pull request by jlloyd-widen**
@tayloramurphy @pnadolny13 

## Checklist

- [ x] Add/update the file in the appropriate folder (`/taps` or `/targets`). The name of the file should match the name of the tap. If there is already one, add a descriptor to the name such as `-search`.
- [ x] Add/update the PNG logo image in `/assets/logos/<taps or targets>`. The image name must match the YAML file name.
- [ x] Tag `@tayloramurphy` or `@pnadolny13` to flag it for review. Or post to the [#hub](https://meltano.slack.com/archives/C01UGBSJNG5) channel on Meltano slack.

## Reviewer Checklist

- [ ] Validate file against JSON Schema. https://www.jsonschema.net/home is an option.
- [ ] Build website locally and validate everything works.

### CODE REVIEW

File: _data/meltano/extractors/tap-gainsightpx/Widen.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  description: 'True' to enable schema flattening and automatically expand nested properties.
+- name: flattening_max_depth
+  label: Flattening Max Depth
+  description: The max depth to flatten schemas.
+domain_url: https://support.gainsight.com/PX/API_for_Developers/02Usage_of_Different_APIs/Work_with_the_Gainsight_PX_REST_API
+maintenance_status: active
+keywords:
+- api
+- gainsight
+- tap
```
Comment (edgarrmondragon) [inline _data/meltano/extractors/tap-gainsightpx/Widen.yml:60]:
```suggestion
- tap
- meltano_sdk
```

**Review by edgarrmondragon — CHANGES_REQUESTED** [review]

File: _data/meltano/extractors/tap-gainsightpx/Widen.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
+name: tap-gainsight
+description: User tracking in web apps
+label: GainsightPX
+namespace: tap_gainsightpx
+logo_url: /assets/logos/extractors/gainsightpx.png
+variant: Widen
```
Comment (tayloramurphy) [inline _data/meltano/extractors/tap-gainsightpx/Widen.yml:6]:
```suggestion
variant: widen
```

**Review by tayloramurphy — CHANGES_REQUESTED** [review]
@jlloyd-widen can you also add an entry in https://github.com/jlloyd-widen/hub/blob/gainsightpx/_data/default_variants.yml#L167 to add tap-gainsightpx? It looks like we don't have that on our checklist currently, but the site won't build without it. Sorry about that!

**Comment by jlloyd-widen** [comment]
> @jlloyd-widen can you also add an entry in https://github.com/jlloyd-widen/hub/blob/gainsightpx/_data/default_variants.yml#L167 to add tap-gainsightpx? It looks like we don't have that on our checklist currently, but the site won't build without it. Sorry about that!

@tayloramurphy Done

File: _data/meltano/extractors/tap-gainsightpx/Widen.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+repo: https://github.com/Widen/tap-gainsightpx
+capabilities:
+- catalog
+- state
+- discover
+- about
+- stream-maps
+- schema-flattening
+settings_group_validation:
+- api_url
```
Comment (pnadolny13) [inline _data/meltano/extractors/tap-gainsightpx/Widen.yml:17]:
```suggestion
- - api_url
  - client_id
  - client_secret
  - refresh_token
  - start_date
```

@jlloyd-widen this key is a list of lists so it needs to be nested one more level deep.

File: _data/meltano/extractors/tap-gainsightpx/Widen.yml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  description: The latest record date to sync (inclusive '<='). ISO format.
+- name: stream_maps
+  label: Stream Maps
+  description: Config object for stream maps capability. For more information check out [Stream Maps](https://sdk.meltano.com/en/latest/stream_maps.html).
+- name: stream_map_config
+  label: Stream Map Config
+  description: User-defined config values to be used within map expressions.
+- name: flattening_enabled
+  label: Flattening Enabled
+  description: 'True' to enable schema flattening and automatically expand nested properties.
```
Comment (pnadolny13) [inline _data/meltano/extractors/tap-gainsightpx/Widen.yml:51]:
```suggestion
  description: "'True' to enable schema flattening and automatically expand nested properties."
```
@jlloyd-widen the single quotes were causing a yaml syntax error. Popping them in double quotes looks to fix it 👍 .

**Review by pnadolny13 — CHANGES_REQUESTED** [review]
@jlloyd-widen thanks for the PR! I left a couple more comments but other than those it looks good to go.

**Review by pnadolny13 — APPROVED** [review]

**Comment by pnadolny13** [comment]
@jlloyd-widen looks good I'll merge it in - thanks again for the PR and the quick fixes!

**Pull request merged**